python/prima/_nonlinear_constraints.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+        
+def process_single_nl_constraint(nlc):
+    '''
+    The Python interfaces receives the constraints as lb <= constraint(x) <= ub, 
+    but the Fortran backend expects the nonlinear constraints to be constr(x) <= 0.
+    Thus a conversion is needed.
+    
+    This function will take a NonlinearConstraint and return a new function which incorporates
+    the lower and upper bounds in such a way as to satisfy the expectation of the Fortran
+    backend. This new function first calls the original function from the provided NonlinearConstraint
+    and obtains the values, and then passes those values to a second function which combines
+    them with the lower and upper bounds as appropriate. The main work here is creating that
+    second function.
+    
+    In general to convert lb <= constraint(x) <= ub to newconstraint(x) <= 0, all we need to do
+    is define newconstraint(x) as [constraint(x) - ub, lb - constraint(x)]. There are some details
+    regarding the type and content of lb and ub as detailed below.
+    
+    
+    The upper and lower bounds could be either scalars or vectors, and so we have 4 possible
+    cases to take into account:
+    
+    1. Both lb and ub are scalars
+        In this case we have a further 4 cases!
+        
+        1a. lb == -inf and ub == inf
```

Comment:
1. `lb == -inf` and `ub == inf` is fine. It just says that nothing is imposed for this constraint. We should remove this constraint in the preprocessing, but should not raise an error. Similar for bound and linear constraints. 

2. Let us write `lb > -inf` and `ub < inf` rather than `lb != -inf` and `ub != inf` even in comments. They are not equivalent if `lb` or `ub` containts NaN. (BTW, `lb` and `ub` should not contain NaN anymore after preprocessing.)

Reply:
1. Should we issue a warning in this case?
2. With the latest version of the code this is no longer an issue (see https://github.com/libprima/prima/pull/124#issuecomment-2003173103)

Reply:
> Should we issue a warning in this case?

We may raise a warning if some nonlinear constraints are trivial and removed. But, to be consistent, we have to do the same for linear constraints as well. 

No need for bound constraints, since `lb[i] == -inf && ub[i] == inf` for some `i` is  quite normal for bound constraints.